test/clients/inst_prepdisk_test.rb

```diff
@@ -39,17 +39,36 @@
       allow(storage).to receive(:prepend_rootprefix).with("/proc").and_return "/dest/proc"
       allow(storage).to receive(:prepend_rootprefix).with("/sys").and_return "/dest/sys"
       allow(Yast::SCR).to receive(:Execute).and_return(true)
+      allow(Yast::Mode).to receive(:update).and_return(mode == :update)
     end
 
-    it "uses the destination directory to mount and prepare the result" do
-      expect(storage).to receive(:rootprefix=).with("/dest")
-      client.run
+    context "in normal mode" do
+      let(:mode) { :normal }
+
+      it "uses the destination directory to mount and prepare the result" do
+        expect(storage).to receive(:rootprefix=).with("/dest")
+        client.run
+      end
+
+      it "commits all libstorage pending changes" do
+        expect(storage).to receive(:calculate_actiongraph)
+        expect(storage).to receive(:commit)
+        client.run
+      end
     end
 
-    it "commits all libstorage pending changes" do
-      expect(storage).to receive(:calculate_actiongraph)
-      expect(storage).to receive(:commit)
-      client.run
+    context "in update mode" do
+      let(:mode) { :update }
+
+      it "does not commit anything" do
+        expect(client).not_to receive(:commit)
```

Comment:
`InstPrepdisk#commit` is a protected method. I don't see a good reason to mention it in the test. Checking for `StorageManager#commit` should be enough.